Bug fixes for LevelZero initialization, overlapping ranks, and geopmd shutdown

bgeltz opened the pull request:
- Fixes #2801.
- Fixes intel-innersource/applications.analyzers.geopm.geopm#51.
- Fixes #2803.

bgeltz committed:
Fix exception propagataion issue in Profile

- m_process is initialized to -1.  It will never
  be -1 in an exception case as it will be set
  in init_prof_comm().

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Use constructor attribute to enable L0

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Use the enum values for zeInit

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo commented on service/src/LevelZero.cpp:
```diff
@@ -15,6 +15,11 @@
 
+static void __attribute__((constructor)) geopm_levelzero_init(void)
+{
+    setenv("ZES_ENABLE_SYSMAN", "1", 1);
```
this can be within an

```
#ifdef LEVEL_ZERO
```
clause

bgeltz committed:
Quit the event loop on shutdown

- Otherwise it hangs, and systemd will eventually
  send it a SIGKILL.
- Fixes #2803.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo commented on src/Profile.cpp:
```diff
@@ -155,7 +155,7 @@ namespace geopm
         }
         catch (const Exception &ex) {
-            if (m_process < 0) {
+            if (m_process >= 0) {
```
m_process is set to a the result of comm_rank() in init_proc_comm() on line 140.  This means that m_process will always be non-negative if we enter the try statement.  I'm not sure why we would condition the warning message based on m_process being positive.
